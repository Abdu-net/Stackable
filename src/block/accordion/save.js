/**
 * Internal dependencies
 */
import createStyles from './style'
import { showOptions } from './util'
import SVGArrowIcon from './images/arrow.svg'

/**
 * External dependencies
 */
import classnames from 'classnames'
import { BlockContainer } from '~stackable/components'
import { withBlockStyles, withUniqueClass } from '~stackable/higher-order'
import { createVideoBackground, hasBackgroundOverlay } from '~stackable/util'

/**
 * WordPress dependencies
 */
import { InnerBlocks, RichText } from '@wordpress/block-editor'
import { applyFilters } from '@wordpress/hooks'
import { Fragment } from '@wordpress/element'
import { compose } from '@wordpress/compose'

// Accessibility: https://www.w3.org/TR/wai-aria-practices/examples/accordion/accordion.html
const save = props => {
	const { className } = props
	const {
		design = 'basic',
		shadow = 3,
		titleTag = '',
		title = '',
		openStart = false,
		showArrow = true,
		onlyOnePanelOpen = false,
	} = props.attributes

	const show = showOptions( props )

	const mainClasses = classnames( [
		className,
		'ugb-accordion--v2',
		`ugb-accordion--design-${ design }`,
	], applyFilters( 'stackable.accordion.mainclasses', {
		'ugb-accordion--open': openStart,
		'ugb-accordion--single-open': onlyOnePanelOpen,
	}, props ) )

	const itemClasses = classnames( [
		'ugb-accordion__item',
	], applyFilters( 'stackable.accordion.itemclasses', {
		'ugb--has-background-overlay': show.containerBackground && hasBackgroundOverlay( 'container%s', props.attributes ),
	}, props ) )

	const headingClasses = classnames( [
		'ugb-accordion__heading',
	], applyFilters( 'stackable.accordion.headingclasses', {
		[ `ugb--shadow-${ shadow }` ]: show.headerBackground && shadow !== 3,
		'ugb--has-background-overlay': show.headerBackground && hasBackgroundOverlay( 'container%s', props.attributes ),
	}, design, props ) )

	return (
		<BlockContainer.Save className={ mainClasses } blockProps={ props } aria-expanded={ openStart ? 'true' : 'false' } render={ () => (
			<Fragment>
				<div className={ itemClasses }>
					{ show.containerBackground && createVideoBackground( 'container%s', props ) }
					<div className={ headingClasses }
						role="button"
						tabIndex="0"
					>
						{ show.headerBackground && createVideoBackground( 'container%s', props ) }
						<RichText.Content
							tagName={ titleTag || 'h4' }
							className="ugb-accordion__title"
							role="heading"
							aria-level="3"
							value={ title }
						/>
						{ showArrow &&
							<SVGArrowIcon className="ugb-accordion__arrow" width="20" height="20" />
						}
					</div>
					<div className="ugb-accordion__content" role="region">
						<div className="ugb-accordion__content-inner">
							<InnerBlocks.Content />
						</div>
					</div>
				</div>
			</Fragment>
		) } />
	)
}

export default compose(
	withUniqueClass,
	withBlockStyles( createStyles ),
)( save )
