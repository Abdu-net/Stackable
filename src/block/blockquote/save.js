import { applyFilters } from '@wordpress/hooks'
import classnames from 'classnames'
import { QUOTE_ICONS } from './quotes'
import { RichText } from '@wordpress/editor'

const save = props => {
	const { className } = props
	const {
		color,
		text,
		quoteColor,
		backgroundColor,
		backgroundImageURL,
		backgroundOpacity = 5,
		fixedBackground,
		quotationMark = 'round-thin',
		quotationSize = 70,
		align,
		contentWidth,
		design = 'plain',
		borderRadius = 12,
		shadow = 3,
	} = props.attributes

	const designHasBackground = [ 'basic', 'top-icon' ].includes( design )

	const mainClasses = classnames( [
		className,
		'ugb-blockquote',
		'ugb-blockquote--v2',
		'ugb--background-opacity-' + ( 1 * Math.round( backgroundOpacity / 1 ) ),
		`ugb-blockquote--design-${ design }`,
	], applyFilters( 'stackable.blockquote.mainclasses', {
		'ugb--has-background': designHasBackground && ( backgroundColor || backgroundImageURL ),
		'ugb--has-background-image': designHasBackground && backgroundImageURL,
		[ `ugb--shadow-${ shadow }` ]: designHasBackground && shadow !== 3,
		[ `ugb-content-width` ]: align === 'full' && contentWidth,
		'ugb-blockquote--small-quote': quotationSize < 60,
	}, design, props ) )

	const styles = applyFilters( 'stackable.blockquote.styles', {
		main: {
			'--quote-color': quoteColor ? quoteColor : undefined,
			backgroundColor: designHasBackground && backgroundColor ? backgroundColor : undefined,
			backgroundImage: designHasBackground && backgroundImageURL ? `url(${ backgroundImageURL })` : undefined,
			backgroundAttachment: designHasBackground && fixedBackground ? 'fixed' : undefined,
			'--ugb-background-color': designHasBackground && backgroundImageURL ? backgroundColor : undefined,
			borderRadius: designHasBackground && borderRadius !== 12 ? borderRadius : undefined,
		},
		text: {
			color: color,
		},
	}, design, props )

	return (
		<blockquote
			className={ mainClasses }
			style={ styles.main }>
			{ applyFilters( 'stackable.blockquote.save.output.before', null, design, props ) }
			<div className="ugb-content-wrapper">
				{ QUOTE_ICONS[ quotationMark ].iconFunc( {
					fill: quoteColor,
					width: quotationSize,
					height: quotationSize,
				} ) }
				{ applyFilters( 'stackable.blockquote.save.output',
					<RichText.Content
						tagName="p"
						className="ugb-blockquote__text"
						style={ styles.text }
						value={ text }
					/>,
					design, props
				) }
			</div>
		</blockquote>
	)
}

export default save
