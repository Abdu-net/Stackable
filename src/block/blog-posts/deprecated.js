const deprecatedSchema_1_17_3 = {
	className: {
		type: 'string',
	},
	order: {
		type: 'string',
		default: 'desc',
	},
	orderBy: {
		type: 'string',
		default: 'date',
	},
	categories: {
		type: 'string',
	},
	postsToShow: {
		type: 'number',
		default: 6,
	},
	columns: {
		type: 'number',
		default: 2,
	},
	displayFeaturedImage: {
		type: 'boolean',
		default: true,
	},
	featuredImageShape: {
		type: 'string',
		default: 'landscape',
	},
	displayTitle: {
		type: 'boolean',
		default: true,
	},
	displayDate: {
		type: 'boolean',
		default: true,
	},
	displayCategory: {
		type: 'boolean',
		default: true,
	},
	displayComments: {
		type: 'boolean',
		default: true,
	},
	displayAuthor: {
		type: 'boolean',
		default: true,
	},
	displayExcerpt: {
		type: 'boolean',
		default: true,
	},
	displayReadMoreLink: {
		type: 'boolean',
		default: false,
	},
	readMoreText: {
		type: 'string',
	},
	contentAlign: {
		type: 'string',
	},
	align: {
		type: 'string',
	},
	accentColor: {
		type: 'string',
	},
	design: {
		type: 'string',
		default: 'basic',
	},
	borderRadius: {
		type: 'number',
		default: 12,
	},
	shadow: {
		type: 'number',
		default: 3,
	},

	// Custom CSS attributes.
	customCSSUniqueID: {
		type: 'string',
		default: '',
	},
	customCSS: {
		type: 'string',
		default: '',
	},
	customCSSCompiled: {
		type: 'string',
		default: '',
	},
}

const deprecated = [
	{
		attributes: deprecatedSchema_1_17_3,
		save: () => null,
		migrate: attributes => {
			return {
				...attributes,

				// TODO: migrate to new custom CSS
				// .ugb-blog-posts__category-list -> .ugb-blog-posts__category
				// .ugb-blog-posts__read_more -> .ugb-blog-posts__readmore
				// .ugb-blog-posts__side -> .ugb-blog-posts__content

				numberOfItems: attributes.postsToShow,
			}
		},
	},
]

export default deprecated
