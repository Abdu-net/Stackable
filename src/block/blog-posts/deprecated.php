<?php
/**
 * Handles the migration of old/removed attributes to new attributes.
 */

// Exit if accessed directly.
if ( ! defined( 'ABSPATH' ) ) {
	exit;
}

if ( ! function_exists( 'stackable_block_blog_posts_is_deprecated' ) ) {
	/**
	 * Check whether the blog post block is deprecated based on it's attributes.
	 * 
	 * @param array $attributes Block attributes.
	 * 
	 * @return boolean True if deprecated.
	 * 
	 * @since 2.0
	 */
	function stackable_block_blog_posts_is_deprecated( $attributes ) {
		// Only deprecated if we have any of these attributes.
		return (bool) array_intersect( array_keys( $attributes ), array(
			'accentColor',
			'postsToShow',
			'displayFeaturedImage',
			'displayTitle',
			'displayDate',
			'displayCategory',
			'displayComments',
			'displayAuthor',
			'displayExcerpt',
			'displayReadMoreLink',
			'readMoreText',
			'categories',
		) );
	}
}

if ( ! function_exists( 'stackable_block_blog_posts_migrate_attributes' ) ) {
	/**
	 * Migrate old attributes to new attributes.
	 *
	 * @param array 	$attributes 	Current saved block attributes.
	 * @param string 	$block_name 	The name of the block.
	 *
	 * @return array New set of attributes.
	 * 
	 * @since 2.0
	 */
	function stackable_block_blog_posts_migrate_attributes( $attributes, $block_name ) {
		if ( $block_name !== 'blog-posts' ) {
			return $attributes;
		}

		// Only do for deprecated blocks.
		if ( ! stackable_block_blog_posts_is_deprecated( $attributes ) ) {
			return $attributes;
		}

		$design = ! empty( $attributes['design'] ) ? $attributes['design'] : 'basic';
		$accent_color = ! empty( $attributes['accentColor'] ) ? $attributes['accentColor'] : '';

		if ( array_key_exists( 'postsToShow', $attributes ) ) {
			$attributes['numberOfItems'] = $attributes['postsToShow'];
		}
		if ( array_key_exists( 'design', $attributes ) ) {
			$attributes['categoryHighlighted'] = in_array( $design, array( 'portfolio', 'portfolio2', 'horizontal-card', 'image-card' ) );
		}

		if ( ! array_key_exists( 'metaColor', $attributes ) ) {
			$attributes['metaColor'] = in_array( $design, array( 'portfolio', 'portfolio2' ) ) ? '#ffffff' : '';
			if ( ! empty( $accent_color ) ) {
				if ( in_array( $design, array( 'basic', 'list', 'vertical-card', 'horizontal-card', 'vertical-card2' ) ) ) {
					$attributes['metaColor'] = $accent_color;
				}
			}
		}

		if ( ! array_key_exists( 'categoryColor', $attributes ) ) {
			$attributes['categoryColor'] = '';
			if ( ! empty( $accent_color ) ) {
				if ( in_array( $design, array( 'portfolio', 'portfolio2', 'horizontal-card', 'image-card' ) ) ) {
					$attributes['categoryColor'] = $accent_color;
				}
			}
		}

		$attributes['showImage'] = isset( $attributes['displayFeaturedImage'] ) ? $attributes['displayFeaturedImage'] : '';
		$attributes['showTitle'] = isset( $attributes['displayTitle'] ) ? $attributes['displayTitle'] : '';
		$attributes['showDate'] = isset( $attributes['displayDate'] ) ? $attributes['displayDate'] : '';
		$attributes['showCategory'] = isset( $attributes['displayCategory'] ) ? $attributes['displayCategory'] : '';
		$attributes['showComments'] = isset( $attributes['displayComments'] ) ? $attributes['displayComments'] : '';
		$attributes['showAuthor'] = isset( $attributes['displayAuthor'] ) ? $attributes['displayAuthor'] : '';
		$attributes['showExcerpt'] = isset( $attributes['displayExcerpt'] ) ? $attributes['displayExcerpt'] : '';
		$attributes['showReadmore'] = isset( $attributes['displayReadMoreLink'] ) ? $attributes['displayReadMoreLink'] : false;
		$attributes['readmoreText'] = isset( $attributes['readMoreText'] ) ? $attributes['readMoreText'] : '';

		$attributes['postType'] = 'post';
		$attributes['taxonomyType'] = 'category';
		$attributes['taxonomy'] = ! empty( $attributes['categories'] ) ? $attributes['categories'] : '';

		// Update the custom CSS since the structure has changed.
		$css = $attributes['customCSS'];
		$css = preg_replace( '/\.ugb-blog-posts__category-list/i', '.ugb-blog-posts__category', $css );
		$css = preg_replace( '/\.ugb-blog-posts__read_more/i', '.ugb-blog-posts__readmore', $css );
		$css = preg_replace( '/\.ugb-blog-posts__side/i', '.ugb-blog-posts__content', $css );
		$attributes['customCSS'] = $css;

		$css = $attributes['customCSSCompiled'];
		$css = preg_replace( '/\.ugb-blog-posts__category-list/i', '.ugb-blog-posts__category', $css );
		$css = preg_replace( '/\.ugb-blog-posts__read_more/i', '.ugb-blog-posts__readmore', $css );
		$css = preg_replace( '/\.ugb-blog-posts__side/i', '.ugb-blog-posts__content', $css );
		$attributes['customCSSCompiled'] = $css;

		return $attributes;
	}
	add_filter( 'stackable_block_migrate_attributes', 'stackable_block_blog_posts_migrate_attributes', 10, 2 );
}

if ( ! function_exists( 'stackable_block_blog_posts_migrate_output' ) ) {
	// We use this for old < v2 blocks as unique IDs.
	global $stkbp_unique_id;
	$stkbp_unique_id = 0;

	/**
	 * Creates a fake wrapper around the deprecated blog posts block to make it look like the new blocks.
	 * This is required to make the migrated blocks look good while not yet edited using v2.
	 * Used when the user just updated the plugin but didn't edit the block.
	 * 
	 * @param string $output Block output.
	 * @param array $attributes Migrated attributes.
	 * 
	 * @return string Block output.
	 * 
	 * @since 2.0
	 */
	function stackable_block_blog_posts_migrate_output( $output, $attributes ) {
		if ( ! stackable_block_blog_posts_is_deprecated( $attributes ) ) {
			return $output;
		}

		global $stkbp_unique_id;
		$stkbp_unique_id++;

		$border_radius = '';
		if ( ! empty( $attributes['borderRadius'] ) ) {
			if ( $attributes['design'] === '' || $attributes['design'] === 'basic' || $attributes['design'] === 'list' ) {
				$border_radius = '<style>.ugb-blog-posts-old-' . $stkbp_unique_id . ' .ugb-blog-posts__featured-image img { border-radius: ' . $attributes['borderRadius'] . 'px !important; }</style>';
			} else if ( $attributes['design'] === 'vertical-card2' ) {
				$border_radius = '<style>.ugb-blog-posts-old-' . $stkbp_unique_id . ' .ugb-blog-posts__item, .ugb-blog-posts-old-' . $stkbp_unique_id . ' .ugb-blog-posts__featured-image { border-radius: ' . $attributes['borderRadius'] . 'px !important; }</style>';
			} else {
				$border_radius = '<style>.ugb-blog-posts-old-' . $stkbp_unique_id . ' .ugb-blog-posts__item { border-radius: ' . $attributes['borderRadius'] . 'px !important; }</style>';
			}
		}

		return sprintf( '<div class="ugb-blog-posts ugb-blog-posts-old-%s ugb-blog-posts--columns-%s ugb-blog-posts--v2 ugb-main-block ugb-blog-posts--design-%s %s" style="%s"><div class="ugb-inner-block">%s<div class="ugb-block-content">%s</div></div></div>',
			$stkbp_unique_id,
			$attributes['columns'],
			$attributes['design'],
			$attributes['categoryHighlighted'] ? 'ugb-blog-posts--cat-highlighted' : '',
			isset( $attributes['accentColor'] ) ? '--s-accent-color: ' . $attributes['accentColor'] : '',
			$border_radius,
			$output
		);
	}
	add_filter( 'stackable/blog-posts/edit.output.markup', 'stackable_block_blog_posts_migrate_output', 10, 2 );
}