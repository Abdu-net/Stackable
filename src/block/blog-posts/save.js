/**
 * External dependencies
 */
import { BlockContainer } from '~stackable/components'
import { withBlockStyles, withUniqueClass } from '~stackable/higher-order'
import classnames from 'classnames'

/**
 * Internal dependencies
 */
import createStyles from './style'

/**
 * WordPress dependencies
 */
import { applyFilters } from '@wordpress/hooks'
import { compose } from '@wordpress/compose'

const save = props => {
	const { className, attributes } = props
	const {
		columns = 2,
		design = 'basic',
	} = attributes

	const mainClasses = classnames( [
		className,
		'ugb-blog-posts--v2',
		`ugb-blog-posts--design-${ design }`,
		`ugb-blog-posts--columns-${ columns }`,
	], applyFilters( 'stackable.blog-posts.mainclasses', {
	}, props ) )

	return (
		<BlockContainer.Save className={ mainClasses } blockProps={ props } render={ () => null } />
	)
}

export default compose(
	withUniqueClass,
	withBlockStyles( createStyles ),
)( save )
