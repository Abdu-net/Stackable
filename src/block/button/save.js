import { applyFilters } from '@wordpress/hooks'
import { ButtonEdit } from '@stackable/components'
import classnames from 'classnames'
import { Fragment } from '@wordpress/element'
import { range } from '@stackable/util'

const save = props => {
	const { className, attributes } = props
	const {
		align = 'center',
		cornerButtonRadius,
		buttons = 1,
		design = '',
	} = attributes

	const mainClasses = classnames( [
		className,
		'ugb-button-wrapper',
	], applyFilters( 'stackable.button.mainclasses', {
		[ `ugb-button--align-${ align }` ]: align,
	}, design, props ) )

	const saved = (
		<Fragment>
			{ applyFilters( 'stackable.button.save.output.before', null, design, props ) }
			<div className={ mainClasses }>
				{ range( 1, buttons + 1 ).map( i => {
					const {
						[ `text${ i === 1 ? '' : i }` ]: text,
						[ `size${ i === 1 ? '' : i }` ]: size = 'normal',
						[ `url${ i === 1 ? '' : i }` ]: url,
						[ `design${ i === 1 ? '' : i }` ]: design = 'basic',
						[ `color${ i === 1 ? '' : i }` ]: color,
						[ `textColor${ i === 1 ? '' : i }` ]: textColor,
						[ `icon${ i === 1 ? '' : i }` ]: icon,
						[ `newTab${ i === 1 ? '' : i }` ]: newTab,
					} = attributes

					const buttonClasses = classnames(
						applyFilters( 'stackable.button.buttonclasses', {}, design, i, props )
					)

					return (
						<ButtonEdit.Content key={ i }
							className={ buttonClasses }
							align={ align }
							size={ size }
							url={ url }
							newTab={ newTab }
							color={ textColor }
							text={ text }
							backgroundColor={ color }
							borderRadius={ cornerButtonRadius }
							design={ design }
							icon={ icon }
						/>
					)
				} ) }
			</div>
			{ applyFilters( 'stackable.button.save.output.after', null, design, props ) }
		</Fragment>
	)

	return applyFilters( 'stackable.designs.button.save', saved, props )
}

export default save
