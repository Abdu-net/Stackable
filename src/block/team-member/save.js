import { applyFilters } from '@wordpress/hooks'
import classnames from 'classnames'
import { Fragment } from '@wordpress/element'
import { range } from '@stackable/util'
import { RichText } from '@wordpress/editor'

const save = props => {
	const { className, attributes } = props
	const {
		shapes,
		nameColor,
		posColor,
		desColor,
		columns = 2,
		design = 'basic',
		borderRadius = 12,
		shadow = 3,
		colorOnHover = false,
	} = props.attributes

	const mainClasses = classnames( [
		className,
		'ugb-team-member',
		'ugb-team-member--v2',
		`ugb-team-member--columns-${ columns }`,
		`ugb-team-member--image-${ shapes }`,
		`ugb-team-member--design-${ design }`,
	], {
		'ugb-team-member--color-on-hover': colorOnHover,
	} )

	const itemClasses = classnames( [
		'ugb-team-member__item',
	], {
		[ `ugb--shadow-${ shadow }` ]: design !== 'plain' && shadow !== 3,
	} )

	return (
		<div className={ mainClasses }>
			{ range( 1, columns + 1 ).map( i => {
				// const href = attributes[ `href${i}` ]
				const mediaURL = attributes[ `mediaURL${ i }` ]
				const name = attributes[ `name${ i }` ]
				const position = attributes[ `position${ i }` ]
				const description = attributes[ `description${ i }` ]

				const styles = applyFilters( 'stackable.team-member.itemstyles', {
					item: {
						borderRadius: design !== 'plain' && borderRadius !== 12 ? borderRadius : undefined,
					},
				}, design, i, props )

				const imageComp = mediaURL && (
					<div className="ugb-team-member__image"
						style={ { backgroundImage: mediaURL ? `url(${ mediaURL })` : undefined } }
						data-src={ mediaURL ? mediaURL : undefined }
					/>
				)
				const nameComp = ! RichText.isEmpty( name ) && (
					<RichText.Content
						tagName="h4"
						className="ugb-team-member__name"
						style={ { color: nameColor } }
						value={ name }
					/>
				)
				const positionComp = ! RichText.isEmpty( position ) && (
					<RichText.Content
						tagName="p"
						className="ugb-team-member__position"
						style={ { color: posColor } }
						value={ position }
					/>
				)
				const descriptionComp = ! RichText.isEmpty( description ) && (
					<RichText.Content
						tagName="p"
						className="ugb-team-member__description"
						style={ { color: desColor } }
						value={ description }
					/>
				)
				const comps = {
					imageComp,
					nameComp,
					positionComp,
					descriptionComp,
				}

				return (
					<div className={ itemClasses } style={ styles.item } key={ i }>
						{ applyFilters( 'stackable.team-member.save.output', (
							<Fragment>
								{ imageComp }
								<div className="ugb-team-member__content">
									{ nameComp }
									{ positionComp }
									{ descriptionComp }
								</div>
							</Fragment>
						), design, comps, i, props ) }
					</div>
				)
			} ) }
		</div>
	)
}

export default save
