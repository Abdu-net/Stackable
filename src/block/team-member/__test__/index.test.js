import { name, settings } from '../'
import blockEditableAfterSaveTests from '@stackable/test/shared/block-editable-after-save'
import blockMigrationTests from '@stackable/test/shared/block-migration'
import deprecated from '../deprecated'
import save from '../save'

describe( `${ settings.title } block`, () => {
	const defaultAttributes = {
		columns: 2,
		description3: '',
		name3: '',
		position3: '',
		mediaURL3: '',
	}

	const attributes = {
		columns: 1,
		description2: '',
		description3: '',
		name2: '',
		name3: '',
		position2: '',
		position3: '',
		mediaURL2: '',
		mediaURL3: '',
	}

	// Checks whether adding the block, saving it then refreshing the editor renders the block valid & editable.
	// Checks whether adding the block, changing values, saving it then refreshing the editor renders the block valid & editable.
	blockEditableAfterSaveTests.bind( this )( {
		name,
		settings,
		save,
		deprecated,
		defaultAttributes,
		attributes,
	} )

	// Checks whether saved HTML of older versioned blocks would migrate and remain valid & editable.
	// Checks whether saved HTML of older versioned blocks with changed values, would migrate and remain valid & editable.
	describe( 'Deprecated migration', () => {
		blockMigrationTests.bind( this )( {
			name,
			settings,
			save,
			deprecated,
		} )
	} )
} )
