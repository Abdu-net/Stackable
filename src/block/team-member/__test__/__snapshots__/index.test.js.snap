// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Member block saved block HTML shouldn't change 1`] = `
<BlockContentProvider
  innerBlocks={Array []}
>
  <div
    className="wp-block-ugb-team-member ugb-team-member ugb-team-member--v2 ugb-team-member--columns-1 ugb-team-member--image-shapes ugb-team-member--design-design className"
  >
    <div
      className="ugb-team-member__item ugb--shadow-1"
      style={
        Object {
          "borderRadius": 6,
        }
      }
    >
      <div
        className="ugb-team-member__image"
        data-src="mediaURL1"
        style={
          Object {
            "backgroundImage": "url(mediaURL1)",
          }
        }
      />
      <div
        className="ugb-team-member__content"
      >
        <Unknown
          className="ugb-team-member__name"
          format="string"
          style={
            Object {
              "color": "#123456",
            }
          }
          tagName="h4"
          value="name1"
        />
        <Unknown
          className="ugb-team-member__position"
          format="string"
          style={
            Object {
              "color": "#123456",
            }
          }
          tagName="p"
          value="position1"
        />
        <Unknown
          className="ugb-team-member__description"
          format="string"
          style={
            Object {
              "color": "#123456",
            }
          }
          tagName="p"
          value="description1"
        />
      </div>
    </div>
  </div>
</BlockContentProvider>
`;

exports[`Team Member block saved default block HTML shouldn't change 1`] = `
<BlockContentProvider
  innerBlocks={Array []}
>
  <div
    className="wp-block-ugb-team-member ugb-team-member ugb-team-member--v2 ugb-team-member--columns-2 ugb-team-member--image-circle ugb-team-member--design-basic"
  >
    <div
      className="ugb-team-member__item"
      style={
        Object {
          "borderRadius": undefined,
        }
      }
    >
      
      <div
        className="ugb-team-member__content"
      >
        <Unknown
          className="ugb-team-member__name"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="h4"
          value="Name"
        />
        <Unknown
          className="ugb-team-member__position"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="p"
          value="Position"
        />
        <Unknown
          className="ugb-team-member__description"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="p"
          value="Description for this block. Use this space for describing your block. Any text will do. Description for this block. You can use this space for describing your block."
        />
      </div>
    </div>
    <div
      className="ugb-team-member__item"
      style={
        Object {
          "borderRadius": undefined,
        }
      }
    >
      
      <div
        className="ugb-team-member__content"
      >
        <Unknown
          className="ugb-team-member__name"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="h4"
          value="Name"
        />
        <Unknown
          className="ugb-team-member__position"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="p"
          value="Position"
        />
        <Unknown
          className="ugb-team-member__description"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="p"
          value="Description for this block. Use this space for describing your block. Any text will do. Description for this block. You can use this space for describing your block."
        />
      </div>
    </div>
  </div>
</BlockContentProvider>
`;
